### PR TITLE
Compare binary size with latest `master` build on CI

### PR DESCRIPTION
**Pull request by fire**
This makes it easier to track binary size regressions, as you no longer have to build the editor and/or export templates locally to do so.

@Calinou Can you link the proposal number for your pr?

- [x] Copy the correct numbers for the hard coded values

### CODE REVIEW

**Comment by Calinou** [comment]
I don't think comparing the binary size with hardcoded sizes is the right approach. I want to see on CI if the PR's build is larger than the latest successful `master` build on CI, disregarding official releases. We can't use official releases as a basis here as they're built with different SCons options that lead to much smaller binaries.

**Comment by fire** [comment]
I disagree, we can use the data from for example github actions' 4.2 branch. The idea is that the hardcoded values don't change every commit.

**Comment by Calinou** [comment]
> I disagree, we can use the data from for example github actions' 4.2 branch. The idea is that the hardcoded values don't change every commit.

If we do this, then we don't know anymore if a PR is actually helping reduce Godot's binary size, or is increasing it.

For example:

- `4.2` branch's latest build is 33,000,000 bytes.
- `master` branch's latest build is 34,000,000 bytes.
- PR X decreases binary size by 4,000 bytes over `master`. Compared to `4.2`, PR X appears 996,000 bytes larger.
- PR Y increases binary size by 1,500 bytes over `master`. Compared to `4.2`, PR Y appears 998,500 bytes larger.

**Comment by fire** [comment]
Maybe we can store the last build size on cache?

**Comment by Calinou** [comment]
> Maybe we can store the last build size on cache?

We could have a dedicated artifact for this if we really want to avoid downloading previous builds, but I don't think that's a big issue. Make sure to run `strip` on the binaries beforehand before uploading their size somewhere though.

File: .github/actions/upload-artifact/action.yml [hunk 1]
```diff
@@ -17,3 +17,63 @@ runs:
...
+        do
+          strip "$file"
+        done
+      shell: bash
+
+    - name: Store artifact size
+      shell: bash
+      run: |
+        git_hash=${{ steps.get_commit_hash.outputs.hash }}
+        actual_size=$(du -sb ${{ inputs.path }} | cut -f1)
```
Comment (AThousandShips) [inline .github/actions/upload-artifact/action.yml:37]: `b` is not a valid flag for `du`, what was it intended to do?

Comment (fire) [reply .github/actions/upload-artifact/action.yml:37]: I was trying to look for a flag that allows getting byte size but this isn't a flag on the macos.

Comment (AThousandShips) [reply .github/actions/upload-artifact/action.yml:37]: There's `B` for block size

**Comment by fire** [comment]
This is stuck looking for a way to determine "last" logical commit and downloadd, executing it to calculate binary size.

**Pull request closed without merging**